src/horreum/horreum_client.py

```diff
@@ -62,7 +62,12 @@ async def setup(self):
         """
 
         if self.__credentials:
-            if self.__credentials.username is not None:
+            if self.__credentials.apikey is not None and (self.__client_config.auth_method is None or self.__client_config.auth_method == AuthMethod.API_KEY) :
```

Comment:
If [@lampajr's assertion](https://github.com/Hyperfoil/horreum-client-python/pull/25#discussion_r1832378133) is correct and `self.__client_config` could be `None` here, then this code won't be happy.  (It doesn't look like you actually accepted his suggestion -- you're checking the `auth_method` for `None`, which is good, but you've skipped the check for `__client_config`.)

Is this a problem?  (Should there be a test for this case?)

Reply:
yeah I think `self.__client_config.auth_method is None` should be converted to `self.__client_config is None`

Reply:
>Checking both auth_method and __client_config would probably be good, e.g.:

I don't think that is strictly necessary as if the `__client_config` is NOT None, then the auth_method should be set (and it is an enum), therefore checking `self.__client_config is None or self.__client_config.auth_method == AuthMethod.API_KEY` should be enough

Reply:
my bad. thanks  @webbnh for the sharp eye !!